infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -211,9 +211,9 @@ private void connectAndQueryTopology() throws LogReplicationDiscoveryServiceExce
             try {
                 // Connect to Cluster Manager and Retrieve Topology Info
                 log.info("Connecting to Cluster Manager adapter...");
-                clusterManagerAdapter.connect(this);
+                clusterManagerAdapter.register(this);
```

Comment:
The start is still not being called. So, any implementation of the interface itself will miss running the start logic.

Reply:
It is said in the comment that any implementation of the register should call the start.